src/Runner.Worker/FileCommandManager.cs

```diff
@@ -325,4 +325,280 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
             }
         }
     }
+
+    public sealed class SaveStateFileCommand : RunnerService, IFileCommandExtension
+    {
+        public string ContextName => "state";
+        public string FilePrefix => "save_state_";
+
+        public Type ExtensionType => typeof(IFileCommandExtension);
+
+        public void ProcessCommand(IExecutionContext context, string filePath, ContainerInfo container)
+        {
+            try
+            {
+                var text = File.ReadAllText(filePath) ?? string.Empty;
+                var index = 0;
+                var line = ReadLine(text, ref index);
+                while (line != null)
+                {
+                    if (!string.IsNullOrEmpty(line))
+                    {
+                        var stateName = string.Empty;
+                        var stateValue = string.Empty;
+
+                        var equalsIndex = line.IndexOf("=", StringComparison.Ordinal);
+                        var heredocIndex = line.IndexOf("<<", StringComparison.Ordinal);
+
+                        // Normal style NAME=VALUE
+                        if (equalsIndex >= 0 && (heredocIndex < 0 || equalsIndex < heredocIndex))
+                        {
```

Comment:
can we refactor this and set-env to a shared function? Hopefully we could use that to get back a key and a value, and do what we want with it!

Reply:
Yes, 100% thank you! I wanted to get to a safe place with test coverage before adventuring deeper in C# land, getting to it now ❤️